addons/controllers/antrea/antreaconfig_controller.go

```diff
@@ -50,47 +51,42 @@ type AntreaConfigReconciler struct {
 // the user.
 
 func (r *AntreaConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("antreaconfig", req.NamespacedName)
+	log := r.Log.WithValues("AntreaConfig", req.NamespacedName)
 
-	r.Log.Info("Start reconciliation")
+	log.Info("Start reconciliation")
 
-	// fetch AntreaConfig resource, ignore not-found errors
-	antreaConfig := &cniv1alpha1.AntreaConfig{}
-	if err := r.Client.Get(ctx, req.NamespacedName, antreaConfig); err != nil {
+	addonConfig := &cniv1alpha1.AntreaConfig{}
+	if err := r.Client.Get(ctx, req.NamespacedName, addonConfig); err != nil {
 		if apierrors.IsNotFound(err) {
-			r.Log.Info(fmt.Sprintf("AntreaConfig resource '%v' not found", req.NamespacedName))
+			log.Info(fmt.Sprintf("AntreaConfig resource '%v' not found", req.NamespacedName))
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, err
 	}
 
-	// deep copy AntreaConfig to avoid issues if in the future other controllers where interacting with the same copy
-	antreaConfig = antreaConfig.DeepCopy()
-
-	// get the parent cluster name from owner reference
-	// if the owner reference doesn't exist, use the same name as config CRD
-	clusterNamespacedName := req.NamespacedName
-	cluster := &clusterapiv1beta1.Cluster{}
-	for _, owner := range antreaConfig.OwnerReferences {
-		if owner.Kind == constants.ClusterKind {
-			clusterNamespacedName.Name = owner.Name
-			break
-		}
+	annotations := addonConfig.GetAnnotations()
+	if _, ok := annotations[constants.TKGAnnotationTemplateConfig]; ok {
+		log.Info(fmt.Sprintf("resource '%v' is a config template. Skipping reconciling", req.NamespacedName))
+		return ctrl.Result{}, nil
 	}
 
-	// verify that the cluster related to config is present
-	if err := r.Client.Get(ctx, clusterNamespacedName, cluster); err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Info(fmt.Sprintf("Cluster '%s'not found in '%s'", clusterNamespacedName.Name, clusterNamespacedName.Namespace))
-			return ctrl.Result{}, nil
-		}
+	// deep copy AntreaConfig to avoid issues if in the future other controllers where interacting with the same copy
+	addonConfig = addonConfig.DeepCopy()
 
-		log.Error(err, fmt.Sprintf("unable to fetch cluster '%s' in '%s'", clusterNamespacedName.Name, clusterNamespacedName.Namespace))
-		return ctrl.Result{}, err
+	cluster, err := cutil.GetOwnerCluster(ctx, r.Client, addonConfig, req.Namespace, constants.AntreaDefaultRefName)
+
+	if err != nil {
+		if apierrors.IsNotFound(err) && cluster != nil {
+			log.Info(fmt.Sprintf("'%s/%s' is listed as owner reference but could not be found",
+				cluster.Name, cluster.Namespace))
+			return ctrl.Result{}, err // (TODO) adduarte check if return  ctrl.Result{}, err ot  ctrl.Result{}, nil is best
```

Comment:
typo in the comment ```ot```. For the comment itself, I think you should return ```nil``` for the NotFound case & ```err``` for all other errors

Reply:
will remove comment and return nil for not found and err for all others.